### PR TITLE
Improving Entry page and navigation

### PR DESCRIPTION
**Pull request by hpk42**
Related to concrete user feedback, here is a rework of the entry pages: 

- "Our Mission" is substituted by "What is Autocrypt? What is its status?" with according text. 

- "Features" and "What is autocrypt" pages are merged into new "Project Background" 

- "Implementation status comes directly after spec". 

i think FAQ and "Example Mail Flows" still need more scrutiny, better introduction to not alienate droppers by too much. 

see https://autocrypt.org/en/newcomerentry/ for how it currently looks with this PR.

### CODE REVIEW

**Comment by r10s** [comment]
looks very good to me, much better than the existing wordings.

File: doc/_templates/customindex.html [hunk 1]
```diff
@@ -31,21 +31,26 @@ <h2>Decentralized</h2>
...
-    <h2>Our Mission</h2>
+    <h2>What is Autocrypt? What is its status?</h2>
     <p>
-      End-to-end encrypted e-mail has been around for decades, but has failed to see wide adoption outside of specialist communities.  Autocrypt aims to provide encryption convenient enough for much wider adoption. Through strengthening the e-mail ecosystem step-by-step, we want to increase user autonomy and privacy for everyone.
-    </p>
-    <p>
-      The Autocrypt effort is driven by a diverse group of mail app developers, hackers and researchers who are willing to take fresh approaches, learn from past mistakes, and collectively aim to increase the overall end-to-end encryption of e-mail in the net.
+      Autocrypt is a specification which mail application programmers can implement
+      to offer their users convenient end-to-end encryption for e-mail. The first specification
+      has been released in December 2017 and a few mail programs already have or are heading
```
Comment (dkg) [inline doc/_templates/customindex.html]: native en_US: I'd change "has been" to "was"

File: doc/_templates/customindex.html [hunk 2]
```diff
@@ -31,21 +31,26 @@ <h2>Decentralized</h2>
...
+      to offer their users convenient end-to-end encryption for e-mail. The first specification
+      has been released in December 2017 and a few mail programs already have or are heading 
+      towards Autocrypt-capable releases in spring 2018.  At this point, we are reaching out 
+      to programmers, interested users and researchers to join us in testing and refining 
+      the "Level 1" specification which provides the baseline for e-mail encryption.  Feedback and 
+      questions are welcome, see the <a href="{{ pathto('contact') }}">Contact Page</a>.
     </p>
     <!-- say something about open standards and open source here or rather in the "try/get it now" app page? -->
+    </p>
+    <p>
```
Comment (dkg) [inline doc/_templates/customindex.html]: what's up with these stray `</p>` and `<p>` tags?

File: doc/_templates/customindex.html [hunk 3]
```diff
@@ -31,21 +31,26 @@ <h2>Decentralized</h2>
...
-    </p>
-    <p>
-      The Autocrypt effort is driven by a diverse group of mail app developers, hackers and researchers who are willing to take fresh approaches, learn from past mistakes, and collectively aim to increase the overall end-to-end encryption of e-mail in the net.
+      Autocrypt is a specification which mail application programmers can implement
+      to offer their users convenient end-to-end encryption for e-mail. The first specification
+      has been released in December 2017 and a few mail programs already have or are heading 
+      towards Autocrypt-capable releases in spring 2018.  At this point, we are reaching out 
+      to programmers, interested users and researchers to join us in testing and refining 
+      the "Level 1" specification which provides the baseline for e-mail encryption.  Feedback and 
+      questions are welcome, see the <a href="{{ pathto('contact') }}">Contact Page</a>.
```
Comment (dkg) [inline doc/_templates/customindex.html]: There is a lot of information in this paragraph.  Arguably, it doesn't even have a single topic.  Would it be OK to break it into two paragraphs?

File: doc/contents.rst [hunk 4]
```diff
@@ -9,7 +9,7 @@ and contributors, MUA developers and privacy enthusiasts.
      introduction to why Autocrypt exists and why it is trying
      to achieve more e-mail encryption.
 
-:doc:`features`
+:doc:`principles`
```
Comment (dkg) [inline doc/contents.rst]: The "principles" document doesn't appear to exist any longer.

File: doc/contents.rst [hunk 5]
```diff
@@ -44,7 +44,8 @@ and contributors, MUA developers and privacy enthusiasts.
 .. toctree::
    :hidden:
 
-   features
+   principles
```
Comment (dkg) [inline doc/contents.rst]: "principles"  should be struck here too, right?

File: doc/background.rst [hunk 6]
```diff
@@ -53,3 +59,37 @@ If you are interested to learn more or want to help please :doc:`join our channe
 where we meet next <contact>`.
 
 See :doc:`contents` for an index of all docs and discussion results so far.
+
+
+Design Differences to previous approaches
```
Comment (dkg) [inline doc/background.rst]: Other section headers in this document have consistent capitalization.  Maybe this one should use the same approach?

File: doc/background.rst [hunk 7]
```diff
@@ -1,6 +1,12 @@
-Introducing Autocrypt: Encrypted E-Mail for Humans
+Background
 ==================================================
 
+.. contents::
+
+
```
Comment (dkg) [inline doc/background.rst]: extra whitespace line seems unnecessary

**Comment by dkg** [comment]
I agree, this is a significant improvement over the current site.

I added a bunch of nit-picks in commentary that i think would make it even better, but i'd be fine with a merge even without those nits addressed, since we can address them later.

**Pull request merged**